pkg/kv/postgres/store.go

```diff
@@ -152,26 +153,45 @@ func parseStoreConfig(runtimeParams map[string]string, pgParams *kvparams.Postgr
 }
 
 // setupKeyValueDatabase setup everything required to enable kv over postgres
-func setupKeyValueDatabase(ctx context.Context, conn *pgxpool.Conn, table string, partitionsAmount int) error {
+func setupKeyValueDatabase(ctx context.Context, conn *pgxpool.Conn, table string, partitionsAmount int) (err error) {
+	var aid string
+	aid, err = generateAdvisoryLockID("lakefs:" + table)
+	if err != nil {
+		return err
+	}
+
+	// This will wait indefinitely until the lock can be acquired.
+	_, err = conn.Exec(ctx, `SELECT pg_advisory_lock($1)`, aid)
```

Comment:
@nopcoder can we add a lock_timeout and provide a detailed error message.
We should want to know when a lock was acquired and not released - and add details of required action on the event of.

Reply:
@N-o-Z This is the same technique we used when `go-migrate` was used. (https://github.com/golang-migrate/migrate/blob/dc26c41ac9dac4a1023c87723ffda644e41af1fe/database/pgx/pgx.go#L226)
The lock timeout is specified in the the database configuration (lock_timeout) the default is unlimited.